src/Http/RequestFactory.php

```diff
@@ -60,25 +60,59 @@ public function setProxy($proxy)
 	 */
 	public function createHttpRequest()
 	{
+		$remoteAddr = isset($_SERVER['REMOTE_ADDR']) ? $_SERVER['REMOTE_ADDR'] : NULL;
+		$remoteHost = isset($_SERVER['REMOTE_HOST']) ? $_SERVER['REMOTE_HOST'] : NULL;
+
+		// use real client address and host if trusted proxy is used
+		$usingTrustedProxy = FALSE;
+		foreach ($this->proxies as $proxy) {
+			if (Helpers::ipMatch($remoteAddr, $proxy)) {
+				$usingTrustedProxy = TRUE;
+				break;
+			}
+		}
```

Comment:
This foreach could be rewritten (to avoid variable default + if + break) as such:
```php
$trustedProxies = count(array_filter($this->proxies, function ($proxy) use ($remoteAddr) {
    return Helpers::ipMatch($remoteAddr, $proxy);
});
```

and the check below as:

```php
if ($trustedProxies) {
```